pkg/client/unversioned/deployment.go

```diff
@@ -94,12 +94,13 @@ func (c *deployments) Update(deployment *extensions.Deployment) (result *extensi
 }
 
 // Watch returns a watch.Interface that watches the requested deployments.
-func (c *deployments) Watch(label labels.Selector, field fields.Selector, resourceVersion string) (watch.Interface, error) {
+func (c *deployments) Watch(label labels.Selector, field fields.Selector, opts api.ListOptions) (watch.Interface, error) {
 	return c.client.Get().
 		Prefix("watch").
 		Namespace(c.ns).
 		Resource("deployments").
-		Param("resourceVersion", resourceVersion).
+		Param("resourceVersion", opts.ResourceVersion).
```

Comment:
What you have:

```
 	return c.client.Get().
 		Prefix("watch").
 		Namespace(c.ns).
 		Resource("deployments").
		Param("resourceVersion", opts.ResourceVersion).
		TimeoutSeconds(TimeoutFromListOptions(opts)).
 		LabelsSelectorParam(label).
 		FieldsSelectorParam(field).
```

can be represented as:

```
 	return c.client.Get().
 		Prefix("watch").
 		Namespace(c.ns).
 		Resource("deployments").
		VersionedParams(opts, api.Scheme).
 		LabelsSelectorParam(label).
 		FieldsSelectorParam(field).
```

or

can be represented as:

```
        opts.FieldSelector = field
        opts.LabelSelector = label
 	return c.client.Get().
 		Prefix("watch").
 		Namespace(c.ns).
 		Resource("deployments").
		VersionedParams(opts, api.Scheme)
```

VersionedParams converts an unversioned ListOptions object into versioned query params.